specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+    /// function call. It allows you to attach any Fungible or Non Fungible Token in a call to a
+    /// receiver contract.
+    ///
+    /// Requirements:
+    /// * Caller of the method must attach a deposit of 1 yoctoⓃ for security
+    ///   purposes
+    /// * Contract MUST panic if called by someone other than token owner or,
+    ///   if using Approval Management, one of the approved accounts
+    /// * The receiving contract must implement `mt_on_transfer` according to the
+    ///   standard. If it does not, MultiToken contract's `mt_resolve_batch_transfer` MUST deal
+    ///   with the resulting failed cross-contract call and roll back the transfer.
+    /// * Contract MUST implement the behavior described in `mt_resolve_batch_transfer`
+    /// * `approval_id` is for use with Approval Management extension, see
+    ///   that document for full explanation.
+    /// * If using Approval Management, contract MUST nullify approved accounts on
+    ///   successful transfer.
+    ///
+    /// Arguments:
+    /// * `receiver_id`: the valid NEAR account receiving the token.
+    /// * `token_ids`: the tokens to transfer
+    /// * `amounts`: the amount of tokens to transfer for corresponding token_id
+    /// * `approval_ids`: expected approval IDs. A number smaller than
+    ///    2^53, and therefore representable as JSON. See Approval Management
+    ///    standard for full explanation. Must have same length as token_ids
+    /// * `memo` (optional): for use cases that may benefit from indexing or
+    ///    providing information for a transfer.
+    /// * `msg`: specifies information needed by the receiving contract in
+    ///    order to properly handle the transfer. Can indicate both a function to
+    ///    call and the parameters to pass to that function.
+
+    fn mt_batch_transfer_call(
+        &mut self,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+        memo: Option<String>,
+        msg: String,
+    ) -> PromiseOrValue<Vec<U128>>;
+
+    /// Get the balance of an an account given token_id. For fungible token returns back amount, for
+    /// non fungible token it returns back constant 1.
+    fn mt_balance_of(&self, owner_id: AccountId, token_id: TokenId) -> U128;
+
+    /// Get the balances of an an account given token_ids. For fungible token returns back amount, for
+    /// non fungible token it returns back constant 1. returns vector of balances corresponding to token_ids
+    /// in a 1-1 mapping
+    fn mt_balance_of_batch(&self, owner_id: AccountId, token_ids: Vec<TokenId>) -> Vec<U128>;
+
+    /// Returns the total supply of the token in a decimal string representation given token_id.
+    fn mt_total_supply(&self, token_id: TokenId) -> U128;
+
+    // Returns the total supplies of the tokens given by token_ids in a decimal string representation.
+    fn mt_total_supply_batch(&self, token_ids: Vec<TokenId>) -> Vec<U128>;
```

Comment:
Why a decimal string representation and not the raw `amount` values?  If I have to provide `amount` using the precision from contract metadata, I would expect to have to do the same for other values the smart contract returns, otherwise I need to handle two different `bases` depending on where data came from as a developer interacting with these contracts -- for example, if I want to send the _entire supply_ of a token from one account to another, I will need to use `mt_total_supply` to get the decimal representation, then I will need to _convert it_ it by the contract metadata precision before I can pass the value into a call to transfer that supply amount.  Feels like all of the methods should return `raw` and it should be expected that anyone interacting with the contract needs to use the metadata that the contract exposes to convert the raw values.

Reply:
this actually just a mistake here, mt_total_supply should be the raw value here . Nice catch!  i'll bring this up to just confirm everyones on the same page in next meeting.